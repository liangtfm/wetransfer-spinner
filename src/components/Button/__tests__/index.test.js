import React from "react";
import { render, screen } from "@testing-library/react";

import { Button, ActiveButton, CancelButton } from "..";

describe("Button", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Button>Button Text</Button>);

    expect(asFragment(<Button>Button Text</Button>)).toMatchSnapshot();
    expect(screen.getByRole("button")).toHaveTextContent(/Button Text/i);
  });

  it("renders ActiveButton correctly", () => {
    const { asFragment } = render(<ActiveButton>Active Text</ActiveButton>);

    expect(
      asFragment(<ActiveButton>Active Text</ActiveButton>)
    ).toMatchSnapshot();
    expect(screen.getByRole("button")).toHaveTextContent(/Active Text/i);
  });

  it("renders CancelButton correctly", () => {
    const { asFragment } = render(<CancelButton>Cancel Text</CancelButton>);

    expect(
      asFragment(<CancelButton>Cancel Text</CancelButton>)
    ).toMatchSnapshot();
    expect(screen.getByRole("button")).toHaveTextContent(/Cancel Text/i);
  });
});
