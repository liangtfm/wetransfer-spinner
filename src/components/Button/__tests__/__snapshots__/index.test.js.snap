// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders ActiveButton correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ sc-gsTCUz gmIHCr caSOrm"
  >
    Active Text
  </button>
</DocumentFragment>
`;

exports[`Button renders CancelButton correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ sc-dlfnbm gmIHCr edDKNS"
  >
    Cancel Text
  </button>
</DocumentFragment>
`;

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ gmIHCr"
  >
    Button Text
  </button>
</DocumentFragment>
`;
